Automattic/wp-calypso
# Plans 2023 : Refactor and typescriptify code

#### Proposed Changes

* Feature strings used by the plans details grid have to be used in the plans comparison grid.
* This PR tries to refactor the code as well as introduce typescript so that the code is more manageable
* Also general cleanup of typescript errors are also performed. 


![image](https://user-images.githubusercontent.com/3422709/211743208-5eb873cb-ea1e-4147-93db-b67a0def64f4.png)


#### Testing Instructions

* Go to `/start/onboarding-2023-pricing-grid/plans`
* Make sure the existing grid works as the shipped code 
* Make sure the title for the comparison grid is showing

#### Pre-merge Checklist





Related to #
